### PR TITLE
Revert "Build loaded models outside inference mode"

### PR DESCRIPTION
**Pull request by glenn-jocher**
Reverts ultralytics/ultralytics#25748

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Reverts the explicit opt-out from PyTorch inference mode when loading models and initializing ReID encoders, while retaining safe reassignment of depth-calibration buffers.

### 📊 Key Changes
- Removes `smart_inference_mode(False)` from `load_checkpoint()` and the PyTorch backend’s `load_model()`.
- Removes the same decorator from `ReID.__init__()`, allowing encoder initialization to follow the caller’s inference-mode context.
- Keeps depth calibration updates implemented with tensor reassignment rather than in-place `fill_`, preserving legal writes and saveable buffers on CUDA.

### 🎯 Purpose & Impact
- Loaded PyTorch models and retained ReID encoders are no longer explicitly constructed outside inference mode.
- Depth calibration continues to replace `cal_a` and `cal_b` with new tensors so calibrated buffers remain writable and saveable.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms your contribution was received, and an Ultralytics engineer will assist with the review. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/nn/backends/pytorch.py [hunk 1]
```diff
@@ -42,7 +42,6 @@ def __init__(
         self.verbose = verbose
         super().__init__(weight, device, fp16)
 
-    @smart_inference_mode(False)  # the loaded module outlives this call, so its weights must not be inference tensors
```
Comment (UltralyticsAssistant) [inline ultralytics/nn/backends/pytorch.py:45]: 💡 **MEDIUM**: Removing this decorator leaves the preloaded-module path unprotected. `BasePredictor.stream_inference()` runs under inference mode and can construct `AutoBackend` with an existing module; when `weight.to(self.device)` changes device or dtype here, the retained module receives tensors allocated in inference mode. Any later training or reset path that enables gradients can then fail because inference tensors cannot be changed to `requires_grad=True` outside inference mode. Keep `load_model()` under `smart_inference_mode(False)`. turn0search1

File: ultralytics/nn/tasks.py [hunk 2]
```diff
@@ -1884,7 +1884,6 @@ def _load():
     return ckpt, file
 
 
-@smart_inference_mode(False)  # the loaded model outlives this call, so its weights must not be inference tensors
```
Comment (UltralyticsAssistant) [inline ultralytics/nn/tasks.py:1887]: 💡 **MEDIUM**: Removing this decorator allows checkpoint loading, fusion, and the final device transfer to run inside an active inference mode. This occurs when a `.pt` model is loaded by `BasePredictor` during `stream_inference()` setup, and also when the ReID callback constructs a retained YOLO encoder. The resulting parameters or buffers can be inference tensors, so later operations that expect a normal reusable model can fail outside inference mode. Restore `smart_inference_mode(False)` on this owner. turn0search1

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Reviewed all four changed files. The calibration change is comment-only, but removing the model-loading guards reintroduces inference-tensor lifetime regressions for both checkpoint and preloaded-module paths. Not LGTM.

<details><summary>💬 Posted 2 inline comments</summary>

- 💡 **MEDIUM** `ultralytics/nn/backends/pytorch.py:45` Removing this decorator leaves the preloaded-module path unprotected. `BasePredictor.stream_inference()` runs under inference mode and can construct `AutoBackend` with an existing module; when `weight.to(self.device)` changes device or dtype here, the retained module receives tensors allocated in inference mode. Any later training or reset path that enables gradients can then fail because inference tensors cannot be changed to `requires_grad=True` outside inference mode. Keep `load_model()` und…
- 💡 **MEDIUM** `ultralytics/nn/tasks.py:1887` Removing this decorator allows checkpoint loading, fusion, and the final device transfer to run inside an active inference mode. This occurs when a `.pt` model is loaded by `BasePredictor` during `stream_inference()` setup, and also when the ReID callback constructs a retained YOLO encoder. The resulting parameters or buffers can be inference tensors, so later operations that expect a normal reusable model can fail outside inference mode. Restore `smart_inference_mode(False)` on this owner. tur…
</details>

**Comment by UltralyticsAssistant** [comment]
🎉 PR [#25768](https://github.com/ultralytics/ultralytics/pull/25768) has been merged—thank you, @glenn-jocher!

> “Great things are done by a series of small things brought together.” — Vincent van Gogh

This PR brings model loading and ReID initialization back in line with the caller’s inference-mode context while preserving safe, saveable depth-calibration buffer updates through tensor reassignment.
